pulp_2to3_migration/app/plugin/iso/repository.py

```diff
@@ -69,12 +70,13 @@ def migrate_to_pulp3(cls, pulp2distributor, repo_version):
         publication = repo_version.publication_set.first()
         if not publication:
             # create publication
-            with FilePublication.create(repo_version, pass_through=True) as publication:
-                manifest = Manifest('PULP_MANIFEST')
-                manifest.write(populate(publication))
-                PublishedMetadata.create_from_file(
-                    file=File(open(manifest.relative_path, "rb")), publication=publication
-                )
+            with WorkingDirectory():
+                with FilePublication.create(repo_version, pass_through=True) as publication:
+                    manifest = Manifest('PULP_MANIFEST')
+                    manifest.write(populate(publication))
+                    PublishedMetadata.create_from_file(
+                        file=File(open(manifest.relative_path, "rb")), publication=publication
+                    )
```

Comment:
Does it make sense to just call `publish` https://github.com/pulp/pulp_file/blob/master/pulp_file/app/tasks/publishing.py#L17?

Great catch btw.

Reply:
It does make sense to do that. The only difference in the code is the logging.